src/components/dex/Trade.vue

```diff
@@ -0,0 +1,523 @@
+        // market order
+        price = this.marketPriceForQuantity(this.side, this.$store.state.orderQuantity);
+      }
+      return price;
+    },
+
+    priceLabel() {
+      if (!this.currentMarket) {
+        return '';
+      }
+
+      return 'GAS'; // this.$t('priceBase', { base: this.currentMarket.baseCurrency });
+    },
+
+    quoteBalanceToolTip() {
+      return 'walletBalanceContractBalance'; // TODO: Fix this
+    },
+
+    quoteHolding() {
+      if (this.currentMarket && this.$store.state.holdings) {
+        const holding = _.find(this.$store.state.holdings, (o) => {
+          return o.assetId === this.currentMarket.quoteAssetId;
+        });
+
+        if (holding) {
+          return holding;
+        }
+      }
+
+      return {
+        symbol: this.currentMarket ? this.currentMarket.quoteCurrency : '',
+        balance: 0,
+        totalBalance: 0,
+        contractBalance: 0,
+      };
+    },
+
+    shouldDisableOrderButton() {
+      if (this.isTradingDisabled) {
+        return true;
+      }
+      if (this.orderType === 'Market') {
+        return !this.$store.state.orderQuantity || this.$isPending('placeOrder');
+      }
+      return !this.$store.state.orderQuantity || !this.$store.state.orderPrice || this.$isPending('placeOrder');
+    },
+
+    total() {
+      try {
+        if (!this.$store.state.orderQuantity) {
+          return 0;
+        }
+
+        return new BigNumber(this.price).multipliedBy(new BigNumber(this.$store.state.orderQuantity));
+      } catch (e) {
+        console.log(e);
+        return 0;
+      }
+    },
+  },
+
+  created() {
+    this.loadHoldings();
+
+    this.actionableHolding = this.quoteHolding;
+  },
+
+  data() {
+    return {
+      actionableHolding: '',
+      orderType: 'Limit',
+      postOnly: false,
+      selectedPercent: null,
+      side: 'Buy',
+    };
+  },
+
+  methods: {
+    confirmOrder() {
+      // TODO: Fill this in with API request
+    },
+
+    loadHoldings() {
+      this.$store.dispatch('fetchHoldings', { done: null });
+    },
+
+    loadHoldingsSilently() {
+      this.$store.dispatch('fetchHoldings', { done: null, isRequestSilent: true });
+    },
+
+    marketPriceForQuantity(side, quantity) {
+      let quantityRemaining = new BigNumber(quantity);
+      let totalMultiple = new BigNumber(0);
+      let book = this.$store.state.orderBook.asks;
+
+      if (side === 'Sell') {
+        book = this.$store.state.orderBook.bids;
+      }
+
+      book.forEach((level) => {
+        const takeQuantity = level.quantity.isGreaterThan(quantityRemaining) ? quantityRemaining : level.quantity;
+        if (quantityRemaining.isLessThanOrEqualTo(0)) {
+          return;
+        }
+        quantityRemaining = quantityRemaining.minus(takeQuantity);
+        totalMultiple = totalMultiple.plus(takeQuantity.multipliedBy(level.price));
+      });
+
+      return (totalMultiple / quantity).toString();
+    },
+
+    percent(value) {
+      if (this.side === 'Buy') {
```

Comment:
`return this.side === 'Buy' ? this.percentForBuy(value) : this.percentForSell(value);`